analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -0,0 +1,156 @@
+                methodDeclaration.Reference.Should().HaveCount(1);
+                methodDeclaration.Reference.Single().Should().BeEquivalentTo(new TextRange { StartLine = 6, EndLine = 6, StartOffset = 8, EndOffset = 14 });
+            });
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Method_PreciseLocation_VB() =>
+            Verify("Method.vb", references =>
+            {
+                references.Select(x => x.Declaration.StartLine).Should().BeEquivalentTo(1, 3, 6);   // class 'Sample' on line 1, method 'Method' on line 3, method 'Go' on line 6
+                var methodDeclaration = references.Single(x => x.Declaration.StartLine == 3);
+                methodDeclaration.Declaration.Should().BeEquivalentTo(new TextRange { StartLine = 3, EndLine = 3, StartOffset = 15, EndOffset = 21 });
+                methodDeclaration.Reference.Should().HaveCount(1);
+                methodDeclaration.Reference.Single().Should().BeEquivalentTo(new TextRange { StartLine = 7, EndLine = 7, StartOffset = 8, EndOffset = 14 });
+            });
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Event_CS() =>
+            Verify("Event.cs", 6, 5, 9, 10);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Field_CS() =>
+            Verify("Field.cs", 4, 3, 7, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Local_CS() =>
+            Verify("Local.cs", 4, 7, 5);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Method_CS() =>
+            Verify("Method.cs", 3, 3, 6);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_NamedType_CS() =>
+            Verify("NamedType.cs", 4, 3, 7, 7); // 'var' and type name on the same line
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Parameter_CS() =>
+            Verify("Parameter.cs", 4, 4, 6, 7);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Property_CS() =>
+            Verify("Property.cs", 4, 3, 7, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Setter_CS() =>
+            Verify("Setter.cs", 4, 6, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_TypeParameter_CS() =>
+            Verify("TypeParameter.cs", 5, 2, 4, 6);
+
+        public void Verify(string fileName, int expectedDeclarationCount, int assertedDeclarationLine, params int[] assertedDeclarationLineReferences) =>
+            Verify(fileName, references =>
+                {
+                    references.Where(x => x.Declaration != null).Should().HaveCount(expectedDeclarationCount);
+                    var declarationWithReferences = references.Single(x => x.Declaration.StartLine == assertedDeclarationLine).Reference;
```

Comment:
These are only references. I would rename this to `declarationReferences` or something similar